src/Bicep.Core.UnitTests/SourceLink/SourceArchiveTests.cs

```diff
@@ -499,58 +523,45 @@ public void DuplicateNamesAfterMunging_ShouldHaveSeparateEntries(
         var cacheRootDirectory = fileExplorer.GetDirectory(CacheRootUri);
-        using var stream = sutFile3 is null ?
-            SourceArchive.PackSourcesIntoStream(entrypointFile.SourceFile.Uri, cacheRootDirectory, entrypointFile, sutFile1, sutFile2) :
-            SourceArchive.PackSourcesIntoStream(entrypointFile.SourceFile.Uri, cacheRootDirectory, entrypointFile, sutFile1, sutFile2, sutFile3);
-
-        SourceArchive sourceArchive = SourceArchive.UnpackFromStream(stream).UnwrapOrThrow();
-
-        var archivedFile1 = sourceArchive.SourceFiles.SingleOrDefault(f => f.Path == expectedPath1);
-        var archivedFile2 = sourceArchive.SourceFiles.SingleOrDefault(f => f.Path == expectedPath2);
-        var archivedFile3 = sourceArchive.SourceFiles.SingleOrDefault(f => f.Path == expectedPath3);
-
-        archivedFile1.Should().NotBeNull($"Couldn't find source file \"{inputBicepPath1}\" in archive");
-        archivedFile2.Should().NotBeNull($"Couldn't find source file \"{inputBicepPath2}\" in archive");
-        if (inputBicepPath3 is not null)
-        {
-            archivedFile3.Should().NotBeNull($"Couldn't find source file \"{inputBicepPath3}\" in archive");
-        }
+        var sourceArchive = sutFile3 is null ?
+            SourceArchive.CreateFor(entrypointFile.SourceFile.Uri, cacheRootDirectory, documentLinks: null, entrypointFile, sutFile1, sutFile2) :
+            SourceArchive.CreateFor(entrypointFile.SourceFile.Uri, cacheRootDirectory, documentLinks: null, entrypointFile, sutFile1, sutFile2, sutFile3);
 
-        archivedFile1!.Path.Should().Be(expectedPath1);
-        archivedFile1.ArchivePath.Should().Be(expectedArchivePath1);
+        var archivedFile1 = sourceArchive.FindSourceFile(expectedPath1);
+        archivedFile1.Metadata.ArchivePath.Should().Be(expectedArchivePath1);
 
-        archivedFile2!.Path.Should().Be(expectedPath2);
-        archivedFile2.ArchivePath.Should().Be(expectedArchivePath2);
+        var archivedFile2 = sourceArchive.FindSourceFile(expectedPath2);
+        archivedFile2.Metadata.ArchivePath.Should().Be(expectedArchivePath2);
 
         if (inputBicepPath3 is not null)
         {
-            archivedFile3!.Path.Should().Be(expectedPath3);
-            archivedFile3.ArchivePath.Should().Be(expectedArchivePath3);
+            var archivedFile3 = sourceArchive.FindSourceFile(expectedPath3!);
+            archivedFile3.Metadata.ArchivePath.Should().Be(expectedArchivePath3);
         }
     }
 
     [TestMethod]
     public void GetSourceFiles_ForwardsCompat_ShouldIgnoreUnrecognizedPropertiesInMetadata()
     {
-        var zip = CreateGzippedTarredFileStream(
+        var sut = CreateSourceArchiveResult(
             (
                 "__metadata.json",
                 @"
                 {
                   ""metadataVersion"": 1,
-                  ""entryPoint"": ""file:///main.bicep"",
+                  ""entryPoint"": ""main.bicep"",
```

Comment:
how does this affect backwards/forwards compat?

Reply:
This doesn't seem related to backward or forward compatibility. it's just a specific way of setting up the test data. If I understand correctly, entryPoint never uses a URI format, so I updated it to use a path instead.